src/main/java/org/folio/circulation/domain/notice/schedule/LoanScheduledNoticeHandler.java

```diff
@@ -250,7 +251,31 @@ public static LoanScheduledNoticeHandler using(Clients clients, DateTime systemT
   }
 
   public boolean noticeIsNotRelevant(ScheduledNotice notice, Loan loan) {
-    return loan.isClosed() || beforeDueDateNoticeIsNotRelevant(notice, loan);
+    if (loan.isClosed()) {
+      return true;
+    }
+    TriggeringEvent triggeringEvent = notice.getTriggeringEvent();
+    if (triggeringEvent == DUE_DATE) {
+      return dueDateNoticeIsNotRelevant(notice, loan);
+    } else if (triggeringEvent == AGED_TO_LOST) {
+      return agedToLostNoticeIsNotRelevant(loan);
+    }
+    log.error("Unexpected triggering event {}", triggeringEvent.getRepresentation());
+    throw new UnsupportedOperationException("Unexpected triggering event: " +
+      triggeringEvent.getRepresentation());
+  }
+
+  private boolean dueDateNoticeIsNotRelevant(ScheduledNotice notice, Loan loan) {
```

Comment:
Please concider logging the reason for irrelevance

Reply:
added

Reply:
I think this method can shortened to something like this:
```
  private boolean dueDateNoticeIsNotRelevant(ScheduledNotice notice, Loan loan) {
    String irrelevanceReason = null;
    
    if (beforeDueDateNoticeIsNotRelevant(notice, loan)) {
      irrelevanceReason = "foo";
    } else if (loan.isDeclaredLost()) {
      irrelevanceReason = "bar";
    } else if (loan.isRenewed()) {
      ...
    } else ...

    if (irrelevanceReason != null) {
      log.warn("Scheduled patron notice {} is irrelevant: {}", notice.getId(), irrelevanceReason);
      return true;
    }
    
    return false;
  }
```

Reply:
Changed this implementation with list of messages